src/client.ts

```diff
@@ -355,6 +390,69 @@ export class Client {
     }
   }
 
+  /**
+   * This Method allows the user to take a screenshot. If no selector is passed
+   * then a full page screenshot is saved.
+   * if no filename is passed, then file is stored with a timestamp as the filename
+   * @param params filename, selector
+   * @returns Filename
+   */
+  public async takeScreenshot(
+    params?: { fileName?: string; selector?: string },
+  ): Promise<string> {
+    if (!this.screenshot_folder || !existsSync(this.screenshot_folder)) {
+      throw new Error("The Screenshot folder is not set or doesn't exist");
+    }
+
+    let fileName = `${generateTimestamp()}.jpg`;
```

Comment:
I'd keep this out and leave it up to the user, instead of us deciding on the filename if they omit it in the param